components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java

```diff
@@ -114,46 +113,35 @@ public void create(IdentityRequest.IdentityRequestBuilder builder, HttpServletRe
                 registrationRequestProfile = new RegistrationRequestProfile();
             }
 
-            Object obj = jsonData.get(RegistrationRequest.RegisterRequestConstant.REDIRECT_URIS);
+            Object obj = jsonData.get(RegistrationRequest.RegisterRequestConstant.GRANT_TYPES);
+            if (obj != null && obj instanceof JSONArray) {
+                JSONArray grantTypes = (JSONArray) obj;
+                for (int i = 0; i < grantTypes.size(); i++) {
+                    registrationRequestProfile.getGrantTypes().add(grantTypes.get(i).toString());
```

Comment:
Add null checks for getGrantTypes and get(i) methods, similar to below code to avoid NPE and avoid grant type been "null" String.

`if (redirectUri == null || IdentityUtil.isBlank(redirectUri.toString())) {`